### DISCUSSION TITLE
UFS_IC_generator.py Python error on Hera

### DISCUSSION THREAD

**Discussion by grantfirl** [NCAR/ccpp-scm#255] [Cases, Initial Conditions, Forcing] [answered]
This question comes from @wzzheng90 via email:

After running "source scm/etc/Hera_setup_intel.csh" first ( shapely is installed ),
and then run the following command:

`./UFS_IC_generator.py -l 261.51 38.2 -d 201610030000 -i ../../data/raw_case_input/FV3_C96_example_ICs -g ../../data/raw_case_input/FV3_C96_example_ICs -n fv3_model_point_noah -oc
`

The error message shows as follows:
  ```
File "./UFS_IC_generator.py", line 2045
    print 'Tile found: {0}'.format(tile)
                          ^
SyntaxError: invalid syntax
```

**Comment by grantfirl** [comment, marked as answer]
@wzzheng90 This is a python2/3 problem. The UFS_IC_generator.py script as it sits in the repository is not updated to use python3 yet, but the Hera_setup_intel.csh script is loading python3.

This file has been updated to use python3 in #224, but I haven't tested that code sufficiently enough to have it merged into the master. You can try a couple of different things: 

1. Get the new version of UFS_IC_generator.py from the #224 pull request and try it
2. Try an older version of scm/etc/Hera_setup_intel.csh that still loads a python2 environment (like the one before python3 was introduced: https://github.com/NCAR/ccpp-scm/commit/478b8e39ba068c1cc5f7e491160e6387baad8d93#diff-1968ed9b2cf3fa296f74530e35efc7586e41c0ece288f14a529688f3de88ff3a)

**Comment by wzzheng90** [comment]
Hi, Grant,

Thank you very much for your quick response!
I am trying your method 1 (Get the new version of UFS_IC_generator.py from the #224 pull request).
I have a problem with running it.  It requires additional modules such as "fv3_remap" and "xesmf":

    import fv3_remap
ModuleNotFoundError: No module named 'fv3_remap'

I do not find them on Hera.  Could you please inform me where to get them?
Best,

Weizhong

**Comment by wzzheng90** [comment]
Hi, Grant,

Sorry, I see "fv3_remap.py" in the same directory but still do not find "xesmf":

import xesmf
ModuleNotFoundError: No module named 'xesmf'

Thanks,

Weizhong

**Reply by grantfirl** [reply]
Yes, XESMF is another python package that is not in the Hera python configuration that the Hera setup script is using. You can try these instructions (that are copied from [this document](https://docs.google.com/document/d/18LISyhHVBDmKYcMefYfypHNcnEPQ9hOq4XKd6XvulGs/edit#) that I wrote):

After running Hera_setup_intel.csh, load a more recent version of Anaconda:

`module load anaconda/anaconda3-5.3.1`

You can try to load a virtual environment that I set up:

`conda activate /scratch1/BMC/gmtb/Grant.Firl/xesmf_env`

You may get an error about the ‘conda activate’ command can’t be used and be provided with instructions to modify your .bashrc file. Follow those instructions and try to run the ‘conda activate’ command again afterward.

Try to see if you can load the xESMF python package:
`python -c “import xesmf”`

If that returns to the prompt without giving you an error, then you should be in good shape. I gave you access to the document from which these instructions were copied. There is more information there for using the newer version of UFS_IC_generator.py that may be helpful to you.

**Comment by wzzheng90** [comment]
Hi, Grant,

Your instructions are really helpful.  I can be able to set up a virtual environment and run your example with the new version of UFS_IC_generator.py. And also the SCM works well with its data input.   Thank you very much again!

Weizhong

**Reply by grantfirl** [reply]
@wzzheng90 I'm glad to hear that you were able to get it set up and working. I'll be working to make this smoother for users.
